bspump/http/web/server.py

```diff
@@ -462,6 +467,23 @@ def render_form(self, request, errors={}):
         <script>
 
             function submitForm() {{
+                // check for required fields and submit in case of satisfied required fields
+                let isValid = true;
+                let requiredFields = document.getElementById("main-form").querySelectorAll("[required]");
+            
+                requiredFields.forEach(field => {{
+                        if (!field.value.trim()) {{
+                            isValid = false;
+                            field.style.border = "2px solid red"; // Highlight empty fields
+                        }} else {{
+                            field.style.border = ""; // Reset style
+                        }}
+                    }});
+
+                if (!isValid) {{
+                    alert("All required fields must be filled.");
+                    return ;
+                }}
                 document.getElementById("loading").style.display = "block";
```

Comment:
Does FileField crash if the form is submitted when required=False and no file is selected?